Resources/public/js/views/tabs/ez-locationviewlocationstabview.js

```diff
@@ -106,6 +109,29 @@ YUI.add('ez-locationviewlocationstabview', function (Y) {
         },
 
         /**
+         * Tap event handler on the `Swap location` button. It fires the
+         * `swapLocation` event
+         *
+         * @method _swapLocation
+         * @protected
+         * @param {EventFacade} e
+         */
+        _swapLocation: function (e) {
+            /**
+             * Fired when the user clicks on `Swap location` button
+             *
+             * @event swapLocation
+             * @param {eZ.Location} location the content to be swaped
+             * @param {Function} afterSwapLocationCallback callback function that will be called after
+             *                   swaping location
+             */
+            this.fire('swapLocation', {
+                location: this.get('location'),
+                afterSwapLocationCallback: Y.bind(this._afterSwapCallback, this)
```

Comment:
not sure we need a callback from the view in the case of swap Location. I guess we should just redirect to the new Location after swapping.